cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/SpaceDevelopersRequestsPlaner.java

```diff
@@ -18,56 +14,35 @@
 
 /**
  * This class is responsible to build the requests in the context of space developers according to the CfChanges.
- * The class creates the request tasks by implementing the {@link CfChangeVisitor} interface.
  */
-public class SpaceDevelopersRequestsPlaner implements CfChangeVisitor {
+public class SpaceDevelopersRequestsPlaner {
 
     private static final Log log = Log.getLog(SpaceDevelopersRequestsPlaner.class);
 
     private final SpaceDevelopersOperations spaceDevelopersOperations;
     private final List<Mono<Void>> requests;
 
-    private SpaceDevelopersRequestsPlaner(SpaceDevelopersOperations spaceDevelopersOperations) {
+    public SpaceDevelopersRequestsPlaner(SpaceDevelopersOperations spaceDevelopersOperations) {
         this.spaceDevelopersOperations = spaceDevelopersOperations;
         this.requests = new LinkedList<>();
     }
 
     /**
-     * Creates the request for CfNewObject
-     *
-     * @param newObject the CfNewObject to be visited
-     */
-    @Override
-    public void visitNewObject(CfNewObject newObject) {
-    }
-
-    /**
-     * Creates the requests for CfObjectValueChanged
+     * Creates the requests to assign/revoke space developer's permission.
      *
-     * @param objectValueChanged the CfObjectValueChanged to be visited
+     * @param spaceDevelopersOperations the SpaceDevelopersOperations object.
+     * @param spaceDevelopersChange a list with all the Changes found during diff for the specific space developers.
+     * @return Flux of all requests that are required to apply the changes.
+     * @throws NullPointerException if one of the argument is null.
      */
-    @Override
-    public void visitObjectValueChanged(CfObjectValueChanged objectValueChanged) {
-    }
+    public Flux<Void> createSpaceDevelopersRequests(SpaceDevelopersOperations spaceDevelopersOperations,
+                                                           CfContainerChange spaceDevelopersChange) {
```

Comment:
If you make this method static, we can omit the constructor and most of the private fields of this class. Another possibility would be to make the class a singleton.

Reply:
Yes, I made this method as static.